lisa/nic.py

```diff
@@ -147,26 +176,41 @@ def __len__(self) -> int:
         return len(self.nics)
 
     def append(self, next_node: NicInfo) -> None:
-        self.nics[next_node.upper] = next_node
+        self.nics[next_node.name] = next_node
 
     def is_empty(self) -> bool:
         return len(self.nics) == 0
 
     def get_unpaired_devices(self) -> List[str]:
-        return [x.upper for x in self.nics.values() if not x.lower]
+        return [x.name for x in self.nics.values() if not x.pci_device_name]
+
+    def get_pci_nics(self) -> List[str]:
+        return [x.pci_device_name for x in self.nics.values() if x.pci_device_name]
```

Comment:
It should check has_module, instead of check the name.

Reply:
updated